dicomanonymizer/simpledicomanonymizer.py

```diff
@@ -243,14 +243,68 @@ def anonymizeDICOMFile(inFile, outFile, extraAnonymizationRules = None):
     # Store modified image
     dataset.save_as(outFile)
 
+def get_private_tag(dataset, tag):
+    element = dataset.get(tag)
+
+    elementValue = element.value
+    tagGroup = element.tag.group
+    # The element is a private creator
+    if elementValue in dataset.private_creators(tagGroup):
+        creator = {
+            "tagGroup": tagGroup,
+            "creatorName": element.value
+        }
+        privateElement = None
+    # The element is a private element with an asociated private creator
+    else:
+        # Shift the element tag in order to get the creatorTag
+        # 0x1009 >> 8 will give 0x0010
+        creatorTagElement = element.tag.element >> 8
+        creatorTag = pydicom.tag.Tag(tagGroup, creatorTagElement)
+        creatorDataset = dataset.get(creatorTag)
+        creator = {
+            "tagGroup": tagGroup,
+            "creatorName": creatorDataset.value
+        }
+        # Define which offset should be applied to the creator to find
+        # this element
+        # 0x0010 << 8 will give 0x1000
+        offsetFromCreator = element.tag.element - (creatorTagElement << 8)
+        privateElement = {
+            "element": element,
+            "offset": offsetFromCreator
+        }
+
+    return {
+        "creator": creator,
+        "element": privateElement
+    }
+
 def anonymizeDataset(dataset, extraAnonymizationRules = None):
     currentAnonymizationActions = initializeActions()
 
     if extraAnonymizationRules != None:
         currentAnonymizationActions.update(extraAnonymizationRules)
 
+    privateTags = []
+
     for tag, action in currentAnonymizationActions.items():
         action(dataset, tag)
+        try:
+            element = dataset.get(tag)
+        except:
+            print("Cannot get element from tag: ", tag)
+
+        if element and element.tag.is_private:
+            privateTags.append(get_private_tag(dataset, tag))
 
     # X - Private tags = (0xgggg, 0xeeee) where 0xgggg is odd
     dataset.remove_private_tags()
```

Comment:
it would be great to have a general way to prevent the deletion of all private tags.
(e.g. https://github.com/KitwareMedical/dicom-anonymizer/issues/3)

Reply:
I'll add it, but I don't really understand why it could be good to have it. Then, this package shouldn't be named dicom-anonymization but dicom-tags-manipulation.

Reply:
#13